core/src/main/java/io/dekorate/kubernetes/decorator/AddResourcesRequirementDecorator.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2020 The original authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.dekorate.kubernetes.decorator;
+
+import io.dekorate.kubernetes.annotation.Resources;
+import io.fabric8.kubernetes.api.model.ContainerFluent;
+import io.fabric8.kubernetes.api.model.Quantity;
+
+public class AddResourcesRequirementDecorator extends Decorator<ContainerFluent> {
+
+	private static final String CPU = "cpu";
+	private static final String MEMORY = "memory";
+
+	private final Resources resources;
+
+	public AddResourcesRequirementDecorator(Resources resources) {
+		this.resources = resources;
+	}
+
+	@Override
+	public void visit(ContainerFluent container) {
+		container.buildResources().getRequests().put(CPU, new Quantity(resources.requests().cpu()));
```

Comment:
I think that this is not guaranteed to work as built resources are meant  to be immutable. Even if it does work now, its not guaranteed to work in the future.

A better approach is:

`container.editOrCreateResources().addToRequests(CPU,  new  Quantity( resources.requests().cpu());`